### DISCUSSION TITLE
How to analyze the reward curve of RL-Agent

### DISCUSSION THREAD

**Discussion by YueNing** [yasserfarouk/scml#12] [one-shot-open-discussion]
I have changed many parameters,

But the curve always just rises in the first stage of training.
Then it fell down and kept oscillating.

I guess if replay buffer is too small?
![plt_0.png](https://i.loli.net/2021/03/14/tLzEwq1m8j9ek6P.png)

If anyone has experience on it or know how to find the problem, 
That is very helpful.

Others configuration
- observation, current offer of negotiation
- reward, set reward as profit of agent after every world step

The configuration of the world is shown below.

![config.png](https://i.loli.net/2021/03/14/OLzvmJGcUTt6gh2.png)

**Comment by YueNing** [comment]
Fix bug of observation, change some hyperparameters, get good result

**Important parameters** 

Changing below two parameters will greatly change the reward curve.
These two hyperparameters control **exploration** and **exploitation** of agent.

- **ε greedy minimum**, from 0.05 -> 0.005
- **steps of anneal**, 50000 -> 500

Under the new simple scenario, described here https://github.com/yasserfarouk/scml/discussions/11#discussion-3266187
Too many exploration will result in an unstable reward curve.

After 500 steps, the agent learned a very good strategy.
![my_agent_Greedy.png](https://i.loli.net/2021/03/15/WXSJINBGoKbnQqU.png)

**Reply by yasserfarouk** [reply]
Great results 👍 keep it up
